content/blog/2022/jan/gif-sane-playback-control-ffmpegwasm/index.md

```diff
@@ -0,0 +1,183 @@
+
+Manifest V3 will be a requirement from January 2023 onwards. If you want to publish an extension to Chrome Web Store, be aware that only Manifest V3 is accepted currently.
+
+Chrome-based engines are the only ones supporting it, while Firefox is still on Manifest V2. This means that you might have to support two versions of the same extension. [Mozilla is supposed to release its implementation soon][11], but it seems there will be divergences.
+
+For GIFSane though, I couldn't even use Manifest V3. As of January 2022, [it still does not support WASM][12]. Developers who want to build on top of WASM still need to wait. Hence why it is not available on Chrome Web Store.
+
+Besides, background scripts [need to be migrated to service workers.][13] Depending on the implementation, this might be tricky. FFmpeg.wasm, for one, uses `[URL.createObjectURL()][14]` and `document.\*` in its browser version, both not available in workers.
+
+### SharedArrayBuffer & SPECTRE
+
+To make use of multi-threading, applications compiled with WASM require `SharedArrayBuffer`. But, since [SPECTRE][15] (the security vulnerability, not the Bond movie), this has been a minefield for browsers.
+
+All major browsers [disabled it at the start of 2018][16], only re-enabling it in 2020. If you want a website to use this, you need to set `Cross-Origin-Opener-Policy` and `Cross-Origin-Embedder-Policy` headers. [Extensions have to set those as well][17] to be able to use WASM.
+
+The caveat is that, on Firefox, [extensions can't use `SharedArrayBuffer`][18]. Only a few [privileged extensions can][19], which forbids the usage of multi-threaded WASM extensions on Firefox.
+
+### Single-threaded FFmpeg.wasm is possible 🚀
+
+![Photo by Michael Dziedzic on Unsplash](./fixing.jpg)
+<!-- Photo by <a href="https://unsplash.com/@lazycreekimages?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Michael Dziedzic</a> on <a href="https://unsplash.com/s/photos/fix?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Unsplash</a> -->
+
+
+On the bright side: WASM applications can be compiled to single-threaded, and FFMPEG.wasm is no different. It removes the need for `SharedArrayBuffer` with the only caveat being it will be certainly slower.
+
+As of January 2022, there is no "pre-packaged" FFMPEG.wasm available with single-threaded support. But [@jeromewu][20] already pushed the required changes to compile it. This means you will need to get your hands a little bit dirty.
+
+**Build `ffmpeg.wasm-core`**
+
+You can build manually by heading to [https://github.com/ffmpegwasm/ffmpeg.wasm-core][21] and following the instructions. Or, you can use Github Actions in your favour:
+
+- Fork the project: [https://github.com/ffmpegwasm/ffmpeg.wasm-core][22]
+- Enable the Github Actions (just go to the Actions tab)
+- Push a dummy commit
+- There should be two workflows running in your Github Actions: go to `Build FFmpeg.wasm (Single Thread)`
```

Comment:
```suggestion
- There should be two workflows running in your GitHub Actions: go to `Build FFmpeg.wasm (Single Thread)`
```

Maybe use "select `Build FFmpeg...`" instead of go to?